installer.sh

```diff
@@ -0,0 +1,238 @@
+  echo "Environment=\"TRAVIS_WORKER_SELF_IMAGE=travisci/worker:$TRAVIS_WORKER_VERSION\"" >> /etc/systemd/system/travis-worker.service.d/env.conf
+  systemctl daemon-reload
+}
+
+install_travis_worker_service_file
+
+# Pulls down the travis-worker image
+install_travis_worker() {
+  docker pull travisci/worker:$TRAVIS_WORKER_VERSION
+}
+
+install_travis_worker
+
+pull_build_images() {
+  image_mappings_json=$(curl https://raw.githubusercontent.com/travis-infrastructure/terraform-config/master/aws-production-2/generated-language-mapping.json)
+
+  docker_images=$(echo "$image_mappings_json" | jq -r "[.[]] | unique | .[]")
+
+  for docker_image in $docker_images; do
+    docker pull "$docker_image"
+
+    langs=$(echo "$image_mappings_json" | jq -r "to_entries | map(select(.value | contains(\"$docker_image\"))) | .[] .key")
+
+    for lang in $langs; do
+      docker tag $docker_image travis:$lang
+    done
+  done
+
+  declare -a lang_mappings=('clojure:jvm' 'scala:jvm' 'groovy:jvm' 'java:jvm' 'elixir:erlang' 'node-js:node_js')
+
+  for lang_map in "${lang_mappings[@]}"; do
+    map=$(echo $lang_map|cut -d':' -f 1)
+    lang=$(echo $lang_map|cut -d':' -f 2)
+
+    docker tag travis:$lang travis:$map
+  done
+}
+
+if [[ ! -n $SKIP_DOCKER_POPULATE ]]; then
+  pull_build_images
+fi
+
+configure_travis_worker() {
+  TRAVIS_ENTERPRISE_CONFIG="/etc/default/travis-enterprise"
+  TRAVIS_WORKER_CONFIG="/etc/default/travis-worker"
+
+  # Trusty images don't seem to like SSH
+  echo "export TRAVIS_WORKER_DOCKER_NATIVE=\"true\"" >> $TRAVIS_WORKER_CONFIG
+  echo "export AMQP_URI=\"amqp://travis:${TRAVIS_ENTERPRISE_SECURITY_TOKEN:-travis}@${TRAVIS_ENTERPRISE_HOST:-localhost}/travis\"" >> $TRAVIS_WORKER_CONFIG
+  echo "export BUILD_API_URI=\"https://${TRAVIS_ENTERPRISE_HOST:-localhost}/${TRAVIS_ENTERPRISE_BUILD_ENDPOINT:-}/script\"" >> $TRAVIS_WORKER_CONFIG
+  echo "export TRAVIS_WORKER_BUILD_API_INSECURE_SKIP_VERIFY='true'" >> $TRAVIS_WORKER_CONFIG
+  echo "export POOL_SIZE='2'" >> $TRAVIS_WORKER_CONFIG
+  echo "export PROVIDER_NAME='docker'" >> $TRAVIS_WORKER_CONFIG
+  echo "export TRAVIS_WORKER_DOCKER_ENDPOINT='tcp://localhost:4243'" >> $TRAVIS_WORKER_CONFIG
+
+  if [[ -n $TRAVIS_ENTERPRISE_HOST ]]; then
+    echo "export TRAVIS_ENTERPRISE_HOST=\"$TRAVIS_ENTERPRISE_HOST\"" >> $TRAVIS_ENTERPRISE_CONFIG
+  fi
+
+  if [[ -n $TRAVIS_ENTERPRISE_SECURITY_TOKEN ]]; then
+    echo "export TRAVIS_ENTERPRISE_SECURITY_TOKEN=\"$TRAVIS_ENTERPRISE_SECURITY_TOKEN\"" >> $TRAVIS_ENTERPRISE_CONFIG
+  fi
+
+  if [[ -n $TRAVIS_ENTERPRISE_BUILD_ENDPOINT ]]; then
+    echo "export TRAVIS_ENTERPRISE_BUILD_ENDPOINT=\"$TRAVIS_ENTERPRISE_BUILD_ENDPOINT\"" >> $TRAVIS_ENTERPRISE_CONFIG
+  else
+    echo "export TRAVIS_ENTERPRISE_BUILD_ENDPOINT=\"__build__\"" >> $TRAVIS_ENTERPRISE_CONFIG
+  fi
+
+  if [[ -n $TRAVIS_QUEUE_NAME ]]; then
+    echo "export QUEUE_NAME='$TRAVIS_QUEUE_NAME'" >> $TRAVIS_WORKER_CONFIG
+  else
+    echo "export QUEUE_NAME='builds.trusty'" >> $TRAVIS_WORKER_CONFIG
```

Comment:
Should we look at doing the selection on the worker side instead of queue side? (wait, looks like it's not merged yet: https://github.com/travis-ci/worker/pull/424)

Reply:
That would be perfect, if we can do that!